### ISSUE TITLE
A small question about shared-normalization model

### ISSUE THREAD

**Issue by YeDeming** [allenai/document-qa#17]
Hi Christopher,

I tried to read your code. And I am confused about the batch size in shared-normalization model.

In Class GroupedSpanAnswerEncoder's encode function, I find each paragraph with question and answer as a item. When try to calculate the Shared-Normalization in IndependentBoundsGrouped's predict function, we need the all paragraphs about an answer.

So how could you make the all paragraphs(max to 15) of an answer into the same batch, while the batch_size is fixed. 

Thanks a lot for your time!
YeDeming

**Comment by chrisc36** [comment]
During training we sample up to 2 paragraph for each question, and include them in the same batch.

During testing use the top 15 paragraphs, but we don't need to compute the shared-normalization loss since we only need to find the most confident answer across all paragraphs. Therefore during testing we don't require paragraphs from the same question be in the same batch.

**Comment by YeDeming** [comment]
Thanks for your help!

**Comment by YeDeming** [comment]
Hi, Christopher,

I have two another questions:

In docqa/scripts/ablate_triviaqa.py, I find you set the learning rate of Adadelta algorithm as 1.0.
I am a little confused about the high learning rate. Whether we need to adjust in later epochs?

I use your code to train a Sigmoid-Confidence-Method model, and it get  0.5848 para-text-em and 0.5400 question-text-em before epoch 6. Is it in line with expectations?

Thanks a lot for your time!
YeDeming

**Comment by chrisc36** [comment]
Adadelta automatically adjusts the learning rate, so starting at 1 and letting it adjust the rate as needed typically works well and is standard practice. 

Manually reducing the learning rate at later epochs can sometimes help, however in our case we tend to overfit the training before we run into optimization difficulties so I suspect it wouldn't help in our case.
